modules/ebi-metagenomics/hmmer/hmmsearch/main.nf

```diff
@@ -0,0 +1,67 @@
+
+process HMMER_HMMSEARCH {
+    tag "$meta.id"
+    label 'process_single'
+
+    conda "${moduleDir}/environment.yml"
+    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
+        'https://depot.galaxyproject.org/singularity/hmmer:3.4--hdbdd923_1' :
+        'biocontainers/hmmer:3.4--hdbdd923_1' }"
+
+    input:
+    tuple val(meta), path(fasta)
+    path hmm_db
+
+    output:
+    tuple val(meta), path("*.tbl"), emit: tbl
+    path "versions.yml"           , emit: versions
+
+    when:
+    task.ext.when == null || task.ext.when
+
+    script:
+    def args = task.ext.args ?: ''
+    def prefix = task.ext.prefix ?: "${meta.id}"
+
+    // Input could either be a .hmm file, or a list of files from a wildcard that includes a .hmm file and index files from running hmmpress
+    def hmm_file = ""
+    if (!hmm_db instanceof List) {
+        hmm_file = hmm_db
+    }
+    else{
+        for(iter_file in hmm_db) {
+            if (iter_file.getExtension() == 'hmm'){
+                hmm_file = iter_file
+                break
+            }
+        }
+    }
```

Comment:
I don't understand the purpose of the block of code above; if multiple HMMs are passed here in a list, we only keep the first one and then run hmmsearch on that? I am not sure it will work properly for a list in this way, and I don't see an nf-test with this scenario.
This could probably be done with Groovy code outside this module (or another pre-filtering module before this one) and then pass one HMM profile as input in HMMER_HMMSEARCH